osu.Game/Input/KeyBindingStore.cs

```diff
@@ -65,8 +65,8 @@ private void insertDefaults(IEnumerable<KeyBinding> defaults, int? rulesetId = n
         /// <param name="rulesetId">The ruleset's internal ID.</param>
         /// <param name="variant">An optional variant.</param>
         /// <returns></returns>
-        public IEnumerable<KeyBinding> Query(int? rulesetId = null, int? variant = null) =>
-            GetContext().DatabasedKeyBinding.Where(b => b.RulesetID == rulesetId && b.Variant == variant);
+        public IEnumerable<DatabasedKeyBinding> Query(int? rulesetId = null, int? variant = null) =>
+            GetContext().DatabasedKeyBinding.Where(b => b.RulesetID == rulesetId && b.Variant == variant).ToList();

```

Comment:
Is there a special reason this is done? I don't mind overlooking this in this particular case, but I'm of the belief that the most base of methods like this should always be returning the IEnumerable.

Reply:
This is a case where EF warned about not being able to satisfy a specific portion of the query.

Here the `KeyBinding` exposing is running a `Convert` internally, which couldn't be satisfied by `IQueryable`. I hink this isn't applicable any more since I changed it back to IEnumerable/List exposing. Will test and switch back if so.

Reply:
Ah I was referring to the addition of ToList

Reply:
Yeah I'm still not sure if we should be `ToListing` before returning. In this case, the query is already set in stone, so I don't see any harm in doing so (only the benefit of 100% ensuring single enumeration).

That said, as a caller of these functions it really needs to be made clear. This should be returning a `List<>` if it's calling `ToList`, at very least